tools/validate_pip_freeze.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+# Verify that the installed Python modules match the requirements file provided
+# as the argument.
+import logging
+import os.path
+import pkg_resources
+import re
+import subprocess
+import sys
+import urllib.parse
+
+
+_IGNORED_WHITELIST = [
+    'python-apt',  # installed by apt from a deb
+]
+_HINT_RE = re.compile(r'.*validator-hint:( +name=(?P<name>[\w.-]+))?( +version=(?P<version>[\w.+-]+))? *$')
+
+
+def _duplicates(strings: [str]) -> [str]:
+    counts = {}
+    for s in strings:
+        if s not in counts:
+            counts[s] = 0
+        counts[s] += 1
+    return [s for s, v in counts.items() if v > 1]
+
+
+def _extract_info_from_url(line):
+    parsed = urllib.parse.urlparse(line.strip())
+    base = os.path.basename(parsed.path)
```

Comment:
Since this is a git repo, we may want to change `base` to `repo_spec` or something since it contains the SHA too.

Reply:
Done.